### PR TITLE
feat: supported tools for gemini

### LINKED ISSUE #49: Support tools for Gemini PRO

### PR DESCRIPTION
**Pull request by adubovik**
Resolves https://github.com/epam/ai-dial-adapter-vertexai/issues/49

### CODE REVIEW

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-vertexai-pr-78.nightly-test.deltixhub.io
>E2E tests status: success

File: aidial_adapter_vertexai/chat/consumer.py [hunk 1]
```diff
@@ -7,10 +7,22 @@
 
 
 class Consumer(ABC):
+    """
+    Whether the consumer has sent something to the choice or not.
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/chat/consumer.py]: Is this for the is_empty method?

File: aidial_adapter_vertexai/chat/tools.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    def filter_functions(
+        function_call: Literal["auto", "none", "required"] | FunctionChoice,
+        functions: List[Function],
+    ) -> Tuple[bool, List[Function]]:
+        match function_call:
+            case "none":
+                return False, []
+            case "auto":
+                return False, functions
+            case "required":
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/chat/tools.py]: Where does "required" come from?

File: aidial_adapter_vertexai/chat/tools.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                        parameters=func.parameters,
+                        description=func.description,
+                    )
+                    for func in self.functions
+                ]
+            )
+        ]
+
+    def to_gemini_tool_config(self) -> GeminiToolConfig | None:
+        if not self.functions or not self.required:
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/chat/tools.py]: Shouldn't the "required" property control the mode: ANY vs AUTO?

File: aidial_adapter_vertexai/chat/consumer.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_vertexai/chat/consumer.py]: Indeed. Moved the comment.

File: aidial_adapter_vertexai/chat/tools.py (hunk 2 above)
Comment (adubovik) [reply aidial_adapter_vertexai/chat/tools.py]: It's from [OpenAI API docs](https://platform.openai.com/docs/api-reference/chat/create#chat-create-tool_choice).
But it seems like Azure OpenAI doesn't support it. I've removed it.

File: aidial_adapter_vertexai/chat/tools.py (hunk 3 above)
Comment (adubovik) [reply aidial_adapter_vertexai/chat/tools.py]: Good catch. Fixed.

**Review by Oleksii-Klimov — APPROVED** [review]

**Pull request merged**